lnwire/funding_created.go

```diff
@@ -63,10 +80,36 @@ func (f *FundingCreated) Encode(w *bytes.Buffer, pver uint32) error {
 //
 // This is part of the lnwire.Message interface.
 func (f *FundingCreated) Decode(r io.Reader, pver uint32) error {
-	return ReadElements(
+	err := ReadElements(
 		r, f.PendingChannelID[:], &f.FundingPoint, &f.CommitSig,
-		&f.ExtraData,
 	)
+	if err != nil {
+		return err
+	}
+
+	var tlvRecords ExtraOpaqueData
+	if err := ReadElements(r, &tlvRecords); err != nil {
+		return err
+	}
+
+	var (
+		partialSig PartialSigWithNonce
+	)
+	typeMap, err := tlvRecords.ExtractRecords(&partialSig)
+	if err != nil {
+		return err
+	}
+
+	// Set the corresponding TLV types if they were included in the stream.
+	if val, ok := typeMap[PartialSigWithNonceRecordType]; ok && val == nil {
+		f.PartialSig = &partialSig
+	}
+
+	if len(tlvRecords) != 0 {
+		f.ExtraData = tlvRecords
```

Comment:
So this is a pattern left over from when we needed to start storing the opaque TLV for gossip messages: if we needed to encode/decode TLV, then we also set the `ExtraData` field as it might contain TLV keys we don't actually know about.